### PR TITLE
Add PXC innovation lts package tests

### PR DESCRIPTION
**Pull request by panchal-yash**
Add ansible-utils.sh script (for modifying tasksbooks and playbooks) Add pxc-precheck scripts, add exceptions for pxc81

### CODE REVIEW

File: molecule/pxc/pxc-innovation-lts-bootstrap-install/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+  - name: print PXC version Molecule
+    debug: var=pxc_version
+
+  - name: check PXC {{ major_release_version}} version Molecule
+    command: /package-testing/version_check.sh pxc{{ major_release_version}}
+    when: lookup('env', 'check_version') == "yes" or lookup('env', 'check_version') == ""
+  
+  - name: Perform PXC package prechecks
+    shell: chmod +x /package-testing/scripts/pxc-package-prechecks.sh && /package-testing/scripts/pxc-package-prechecks.sh pxc{{ major_release_version}}
```
Comment (eleo007) [inline molecule/pxc/pxc-innovation-lts-bootstrap-install/tasks/main.yml:158]: It is better to put script with exec rights at once than update them during the playbook execution

File: molecule/pxc/pxc-innovation-lts-bootstrap-upgrade/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    template: src=../../../../templates/my_57.j2
+              dest=/root/.my.cnf
+              mode=0640 owner=root group=root
+    when: ansible_os_family == "RedHat"
+
+  # other tests
+# - name: install plugins
+#   command: /package-testing/plugins_test_80.sh pxc
+
+  - name: register PXC version Molecule
```
Comment (eleo007) [inline molecule/pxc/pxc-innovation-lts-bootstrap-upgrade/tasks/main.yml:137]: Why do we need this if we have it in VERSIONS file?

File: scripts/pxc-package-prechecks.sh [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  echo "Illegal product selected!"
+  exit 1
+fi
+
+
+if [[ $1 =~ ^pxc8[0-9]{1}$ ]]; then
+  if [ -f /usr/bin/wsrep_sst_xtrabackup-v2 ]; then
+    echo "File exists: /usr/bin/wsrep_sst_xtrabackup-v2"
+
+    pxc_version=$(mysql --version  | grep Ver | awk '{print$3}')
```
Comment (eleo007) [inline scripts/pxc-package-prechecks.sh:72]: mysqld 

mysql is client 'binary'. The versions should be the same but let's use the mysqld

File: scripts/pxc-package-prechecks.sh [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    pxc_version_trimmed=$(echo $pxc_version | grep -o -E '[0-9]+\.[0-9]+\.[0-9]+')  # Print the first two characters
```
Comment (eleo007) [inline scripts/pxc-package-prechecks.sh:73]: Isn't it 3? Fix comment

File: scripts/pxc-package-prechecks.sh [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        xb_prev_lts_required_version_trimmed_major_version=$(echo $xb_prev_lts_required_version_trimmed |  grep -o -E '[0-9]+\.[0-9]+')  # Print the first two characters
+    else
+        echo "Not PXC81"
+    fi
+
+        # 1. Checking for the xb_this_required_version 
+
+        if [[ $(check_xb_file $xb_this_required_version_trimmed_major_version ; echo $?) -eq 1 ]]; then
+            pxc_extras_xtrabackup_version_this_required=$(/usr/bin/pxc_extra/pxb-$xb_this_required_version_trimmed_major_version/bin/xtrabackup -v |& grep -oP '(?<=xtrabackup version ).*' | awk '{print$1}')
+            pxc_extras_xtrabackup_version_this_required_trimmed=$( echo $pxc_extras_xtrabackup_version_this_required |  grep -o -E '[0-9]+\.[0-9]+\.[0-9]+')  # Print the first two characters
```
Comment (eleo007) [inline scripts/pxc-package-prechecks.sh:109]: Again seems 3 chars.

**Review by eleo007 — APPROVED** [review]

File: molecule/pxc/pxc-innovation-lts-bootstrap-upgrade/tasks/main.yml (hunk 2 above)
Comment (panchal-yash) [reply molecule/pxc/pxc-innovation-lts-bootstrap-upgrade/tasks/main.yml:137]: This step is for checking the installed pxc version 
after this task we have another task where we print value using ansible debug module

File: molecule/pxc/pxc-innovation-lts-bootstrap-install/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    when: ansible_os_family == "RedHat"
+
+  - name: copy .my.cnf with credentials on centos
+    template: src=../../../../templates/my_57.j2
+              dest=/root/.my.cnf
+              mode=0640 owner=root group=root
+    when: ansible_os_family == "RedHat"
+
+  # other tests
+# - name: install plugins
```
Comment (eleo007) [inline molecule/pxc/pxc-innovation-lts-bootstrap-install/tasks/main.yml:143]: Why do we leave it commented out?

File: molecule/pxc/pxc-innovation-lts-common-install/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+    when: lookup('env', 'check_version') == "yes" or lookup('env', 'check_version') == ""
+
+  - name: check that PXC package versions are correct Molecule
+    command: /package-testing/package_check.sh pxc{{ major_release_version}}
+    when: lookup('env', 'check_version') == "yes" or lookup('env', 'check_version') == ""
+
+  - name: Perform PXC package prechecks
+    command: /package-testing/scripts/pxc-package-prechecks.sh pxc{{ major_release_version}}
+    
+# - name: install plugins
```
Comment (eleo007) [inline molecule/pxc/pxc-innovation-lts-common-install/tasks/main.yml:133]: The same here

File: molecule/pxc/pxc-innovation-lts-common-upgrade/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+
+  - name: check that PXC version is correct Molecule
+    command: /package-testing/version_check.sh pxc{{ major_release_version}}
+    when: lookup('env', 'check_version') == "yes" or lookup('env', 'check_version') == ""
+
+  - name: check that PXC package versions are correct Molecule
+    command: /package-testing/package_check.sh pxc{{ major_release_version}}
+    when: lookup('env', 'check_version') == "yes" or lookup('env', 'check_version') == ""
+
+# - name: install plugins
```
Comment (eleo007) [inline molecule/pxc/pxc-innovation-lts-common-upgrade/tasks/main.yml:128]: The same as above

File: package_check.sh [hunk 9]
```diff
@@ -248,8 +252,8 @@ elif [ ${product} = "pxc56" -o ${product} = "pxc57" ]; then
     fi
   fi
 
-elif [ ${product} = "pxc80" ]; then
-  echo "Package check for PXC-80 is not implemented!"
+elif [[ ${product} =~ ^pxc8[1-9]{1}$ ]]; then
```
Comment (eleo007) [inline package_check.sh:255]: Should not we have `^pxc8[0-9]{1}$ ]]` ? Otherwise the test will fail for 80.

**Review by eleo007 — CHANGES_REQUESTED** [review]

File: package_check.sh (hunk 9 above)
Comment (panchal-yash) [reply package_check.sh:255]: I think we don't have pxc package check implemented for pxc 80 so I have started it with 81

**Review by kaushikpuneet07 — APPROVED** [review]
LGTM

**Pull request merged**